op-signer/gen-local-creds.sh

```diff
@@ -0,0 +1,189 @@
+#!/usr/bin/env sh
+
+set -euo pipefail
+
+if [ -z "$TLS_DIR" ]; then
+    SCRIPT_DIR="$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
+    TLS_DIR="$SCRIPT_DIR/tls"
+fi
+
+OPENSSL_IMAGE="alpine/openssl:3.3.3"
+
+USER_UID=$(id -u)
+USER_GID=$(id -g)
+
+CERT_ORG_NAME="OP-Signer Local Org"
+MOD_LENGTH=2048
+
+# File paths
+CA_CERT="$TLS_DIR/ca.crt"
+CA_KEY="$TLS_DIR/ca.key"
+CLIENT_TLS_KEY="tls.key"
+CLIENT_TLS_CSR="tls.csr"
+CLIENT_TLS_CERT="tls.crt"
+CLIENT_PRIVATE_KEY="ec_private.pem"
+CLIENT_OPENSSL_CNF="openssl.cnf"
+
+# Check if we should use Docker (default to true if not set)
+USE_DOCKER=${OP_SIGNER_GEN_TLS_DOCKER:-true}
+
+# Helper function to run openssl commands
+run_openssl() {
+    if [ "$USE_DOCKER" = "true" ]; then
+        docker run --rm \
+            -v "$TLS_DIR:$TLS_DIR" \
+            -u "$USER_UID:$USER_GID" \
+            "$OPENSSL_IMAGE" "$@"
+    else
+        # Check if openssl is available locally
+        if ! command -v openssl &> /dev/null; then
+            echo "Error: OpenSSL is not installed locally. Please install OpenSSL or use Docker by setting OP_SIGNER_GEN_TLS_DOCKER=true"
+            exit 1
+        fi
+        openssl "$@"
+    fi
+}
+
+generate_ca() {
+    local force="$1"
+    [ "$force" = "true" ] || [ ! -f "$CA_CERT" ] || return 0
+
+    echo
+    echo "Generating CA..."
+
+    run_openssl req -newkey "rsa:$MOD_LENGTH" \
+        -new -nodes -x509 \
+        -days 365 \
+        -sha256 \
+        -out "$CA_CERT" \
+        -keyout "$CA_KEY" \
+        -subj "/O=$CERT_ORG_NAME/CN=root"
+}
+
+generate_client_tls() {
+    local hostname="$1"
+    echo
+    echo "Generating client TLS credentials for $hostname..."
+    
+    
+    # Create a directory for this client's credentials
+    local clientDir="$TLS_DIR/$hostname"
+    mkdir -p "$clientDir"
+    
+    # Generate client key
+    echo "Generating client key..."
+    run_openssl genrsa -out "$clientDir/$CLIENT_TLS_KEY" "$MOD_LENGTH"
+
+    local confFile="$clientDir/$CLIENT_OPENSSL_CNF"
+    
+    # Create a config file for the CSR
+    cat > "$confFile" << EOF
+[req]
+distinguished_name=req
+[san]
+subjectAltName=DNS:$hostname
+EOF
+    
+    echo "Generating client certificate signing request..."
+    run_openssl req -new -key "$clientDir/$CLIENT_TLS_KEY" \
+        -sha256 \
+        -out "$clientDir/$CLIENT_TLS_CSR" \
+        -subj "/O=$CERT_ORG_NAME/CN=$hostname" \
+        -extensions san \
+        -config "$confFile"
+    
+    echo "Generating client certificate..."
+    run_openssl x509 -req -in "$clientDir/$CLIENT_TLS_CSR" \
+        -sha256 \
+        -CA "$CA_CERT" \
+        -CAkey "$CA_KEY" \
+        -CAcreateserial \
+        -out "$clientDir/$CLIENT_TLS_CERT" \
+        -days 3 \
+        -extensions san \
+        -extfile "$confFile"
+}
+
+generate_client_key() {
```

Comment:
Can we rename this to 

```suggestion
generate_signing_key() {
```

The secp256k1 isn't actually the client's key, it's the key that will be maintained by op-signer. In the wild this key should never be accessible to the client and op-signer will provide crypto-as-a-service to the client